### PR TITLE
New package: ncgopher-0.4.0

### PR DESCRIPTION
**Pull request by Goorzhel**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

### CODE REVIEW

File: srcpkgs/ncgopher/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'ncgopher'
+pkgname=ncgopher
+version=0.4.0
+revision=1
+build_style=cargo
+hostmakedepends="ncurses-devel openssl-devel pkg-config sqlite-devel"
```
Comment (classabbyamp) [inline srcpkgs/ncgopher/template:6]: the devel packages should be makedepends

**Comment by Goorzhel** [comment]
On my host:
```
❯ git config alias.b
rev-parse --abbrev-ref HEAD
❯ ./xbps-src -a aarch64 pkg $(git b)
<...>
  --- stderr
  thread 'main' panicked at '/builddir/ncgopher-0.4.0/target/aarch64-unknown-linux-gnu/release/build/ncurses-e718d8c075f3805b/out/chtype_size failed: Os { code: 8, kind: Uncategorized, message: "Exec format error" }', /host/cargo/registry/src/github.com-1ecc6299db9ec823/ncurses-5.101.0/build.rs:107:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
=> ERROR: ncgopher-0.4.0_1: do_build: '${make_cmd} build --
```
Let's hope CI behaves differently.

**Comment by classabbyamp** [comment]
ah I think it's trying to run ncursesw6-config from ncurses-devel, which means it should be in both hostmake and make

**Comment by Goorzhel** [comment]
Abby and I found out on IRC that this is https://github.com/jeaye/ncurses-rs/issues/184.

**Comment by classabbyamp** [comment]
can you rebase this?

**Pull request closed without merging**